### ISSUE TITLE
ReadLine.cpp:11:25: error: readline.h: No such file or directory

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cmake-buildscripts#78]
uhm

I think I reported this somewhere but can not find it. Do feel free to close this if it's duplicated. Just want to post some error here so I can search it later.

```
[ 80%] Building CXX object AmberTools/src/cpptraj/src/CMakeFiles/cpptraj.dir/ReadLine.cpp.o
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:11:25: error: readline.h: No such file or directory
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:12:24: error: history.h: No such file or directory
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp: In function ‘char** cpptraj_completion(const char*, int, int)’:
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:62: error: ‘rl_line_buffer’ was not declared in this scope
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:63: error: ‘rl_completion_matches’ was not declared in this scope
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp: In constructor ‘ReadLine::ReadLine()’:
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:70: error: ‘rl_attempted_completion_function’ was not declared in this scope
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp: In member function ‘int ReadLine::GetInput()’:
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:91: error: ‘readline’ was not declared in this scope
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp: In member function ‘void ReadLine::AddHistory(const char*)’:
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:111: error: ‘add_history’ was not declared in this scope
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp: In member function ‘bool ReadLine::YesNoPrompt(const char*)’:
/home/haichit/amber_git/amber/AmberTools/src/cpptraj/src/ReadLine.cpp:122: error: ‘readline’ was not declared in this scope
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-unused-local-typedefs"
make[2]: *** [AmberTools/src/cpptraj/src/CMakeFiles/cpptraj.dir/ReadLine.cpp.o] Error 1
make[1]: *** [AmberTools/src/cpptraj/src/CMakeFiles/cpptraj.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by hainm** [comment]
command
```
~/programs/cmake-3.9.4/bin/cmake ../../amber \
-DPYTHON_EXECUTABLE=$HOME/miniconda3/bin/python \
-DUSE_MINICONDA=False \
-DCMAKE_INSTALL_PREFIX=/home/haichit/amber_git/cmake-reports/stable-install

make install -j8 # got error, so try "make install" to see the actual error.
```

**Comment by multiplemonomials** [comment]
Can you post the full CMake output at least, please?  I will need to know if this is internal or external readline.

**Comment by hainm** [comment]
ok, so this is a new error. Hold on.

**Comment by multiplemonomials** [comment]
Actually, do you mind just taking a full bug report?  I just checked the cpptraj code over and it looked fine which means it's a problem with wherever the readline target is created or imported.

**Comment by hainm** [comment]
yes I will, just hold on.

**Comment by hainm** [comment]
sounds like two kind of errors.

summary below
```
usr/bin/ld: BFD version 2.20.51.0.2-5.36.el6 20100205 internal error, aborting at merge.c line 876 in _bfd_merged_section_offset

/usr/bin/ld: Please report this bug.

collect2: ld returned 1 exit status
make[2]: *** [AmberTools/src/cpptraj/src/xdrfile/libxdrfile.so] Error 1
make[2]: Leaving directory `/home/haichit/amber_git/cmake-reports/issue-78-build'
make[1]: *** [AmberTools/src/cpptraj/src/xdrfile/CMakeFiles/xdrfile.dir/all] Error 2
make[1]: Leaving directory `/home/haichit/amber_git/cmake-reports/issue-78-build'
make: *** [all] Error 2
```

Full: 
[issue-78.txt](https://github.com/Amber-MD/cmake-buildscripts/files/1385187/issue-78.txt)

readline error:
(getting after "make install" 2nd time in the same build).

[makeoutput_after_1st_fail.txt](https://github.com/Amber-MD/cmake-buildscripts/files/1385188/makeoutput_after_1st_fail.txt)

**Comment by multiplemonomials** [comment]
The LD error seems to be some sort of transient race condition / hardware error (!).  See https://sourceware.org/bugzilla/show_bug.cgi?id=16139

**Comment by hainm** [comment]
if non-cmake work, does it count for that? (just naive question).

**Comment by multiplemonomials** [comment]
Huh?  Count for what?  I don't understand.

**Comment by hainm** [comment]
I meant if the non-cmake is ok, why the possible reason only happens for cmake.

**Comment by multiplemonomials** [comment]
About the LD thing, or about the Readline error?

**Comment by hainm** [comment]
the LD thing.

**Comment by multiplemonomials** [comment]
Wait, did it happen multiple times in a row?

**Comment by hainm** [comment]
it's my turn not to understand you. If we are still talking about the below, we should skip it.

> Huh? Count for what? I don't understand.

**Comment by multiplemonomials** [comment]
OK, my point is that the LD error was transient (a race condition) and shouldn't happen again.

**Comment by multiplemonomials** [comment]
This should be fixed by cpptraj#567.

**Comment by multiplemonomials** [comment]
cpptraj has accepted the PR, so this issue is fixed.
